### PR TITLE
Move out default values out of specific OSes

### PR DESCRIPTION
**Pull request by kronos-pbrideau**
I've taken default values than was in debian OS, and changed custom path by OSes.

### CODE REVIEW

**Comment by edestecd** [comment]
Thanks for the work.  I promise to have a look soon.  I have been swamped at work lately...

**Comment by edestecd** [comment]
It looks good overall, I just want to have a closer look at the individual options and what would be changing on existing systems.  I like the way you organized it.  This is exactly the approach the puppetlabs/mysql takes.

File: manifests/params.pp [hunk 1]
```diff
@@ -70,41 +61,23 @@
...
-        'DatabaseMirror'    => ['db.us.clamav.net', 'db.local.clamav.net'],
-      }
     }
+
+    #### Default values OS specific ####
+    $clamd_default_logfile               = '/var/log/clamav/clamd.log'
+    $clamd_default_pidfile               = '/var/run/clamav/clamd.pid'
+    $clamd_default_temporarydirectory    = '/var/tmp'
+    $clamd_default_databasedirectory     = '/var/lib/clamav'
+    $clamd_default_localsocket           = '/var/run/clamav/clamd.sock'
```
Comment (edestecd) [inline manifests/params.pp:75]: localsocket should be this on CentOS 7:  LocalSocket /var/run/clamd.scan/clamd.sock
/var/run/clamav does not exist after installing the package.
Did this work for you?

File: manifests/params.pp [hunk 2]
```diff
@@ -70,41 +61,23 @@
...
-        'DatabaseDirectory' => '/var/lib/clamav',
-        'UpdateLogFile'     => '/var/log/clamav/freshclam.log',
-        'LogSyslog'         => 'yes',
-        'DatabaseOwner'     => 'clam',
-        'DatabaseMirror'    => ['db.us.clamav.net', 'db.local.clamav.net'],
-      }
     }
+
+    #### Default values OS specific ####
+    $clamd_default_logfile               = '/var/log/clamav/clamd.log'
```
Comment (edestecd) [inline manifests/params.pp:71]: logfile should be this on CentOS 7: LogFile /var/log/clamd.scan
/var/log/clamav does not exist after installing the package.

Also, by default on both CentOS 6 and 7 LogSyslog is true, so maybe we need to retain that or
ensure the log dir/file exists, as it appears not to from the package install.

File: manifests/params.pp [hunk 3]
```diff
@@ -70,41 +61,23 @@
...
+    $clamd_default_pidfile               = '/var/run/clamav/clamd.pid'
```
Comment (edestecd) [inline manifests/params.pp:72]: pidfile should be this on CentOS 7: PidFile /var/run/clamd.scan/clamd.pid
/var/run/clamav does not exist after installing the package.
Did this work for you?

File: manifests/params.pp [hunk 4]
```diff
@@ -130,109 +103,124 @@
...
+  $freshclam_default_options = {
+    'AllowSupplementaryGroups' => false,
+    'Bytecode'                 => true,
+    'Checks'                   => '24',
+    'CompressLocalDatabase'    => 'no',
+    'ConnectTimeout'           => '30',
+    'DNSDatabaseInfo'          => 'current.cvd.clamav.net',
+    'DatabaseDirectory'        => $freshclam_default_databasedirectory,
+    'DatabaseMirror'           => ['db.local.clamav.net', 'database.clamav.net'],
+    'DatabaseOwner'            => $user,
```
Comment (edestecd) [inline manifests/params.pp:209]: DatabaseOwner is a seperate user in CentOS 7: DatabaseOwner clamupdate

File: manifests/params.pp [hunk 5]
```diff
@@ -70,41 +61,23 @@
...
+
+    #### Default values OS specific ####
+    $clamd_default_logfile               = '/var/log/clamav/clamd.log'
+    $clamd_default_pidfile               = '/var/run/clamav/clamd.pid'
+    $clamd_default_temporarydirectory    = '/var/tmp'
+    $clamd_default_databasedirectory     = '/var/lib/clamav'
+    $clamd_default_localsocket           = '/var/run/clamav/clamd.sock'
+
+    $freshclam_default_databasedirectory = '/var/lib/clamav'
+    $freshclam_default_updatelogfile     = '/var/log/clamav/freshclam.log'
```
Comment (edestecd) [inline manifests/params.pp:78]: CentOS 7: UpdateLogFile /var/log/freshclam.log
OR syslog(default) OR make this dir with puppet

**Comment by edestecd** [comment]
@kronos-pbrideau sorry for the delay.  I'd like to get this merged soon.  Can you start by doing a rebase?

**Comment by kronos-pbrideau** [comment]
@edestecd Sorry for the delay.  I have a heavily modified profile which call this module, so it works on our servers, but yeah, feel free to set these defaults to where they should be :)

**Comment by edestecd** [comment]
No problem.  I went ahead and resolved the conflicts and updated the defaults for RHEL 7.
What OS and version are you using?

Thank you for your contribution.

**Comment by kronos-pbrideau** [comment]
We're on lot of stuff: Centos 6-7, Debian 7-8 and Ubuntu 14.04, i'll check your changes soon. Thanks :)

**Pull request merged**